etc/profile-a-l/cargo.profile

```diff
@@ -0,0 +1,80 @@
+# Firejail profile for cargo
+# Description: The Rust package manager
+# This file is overwritten after every install/update
+quiet
+# Persistent local customizations
+include cargo.local
+# Persistent global definitions
+include globals.local
+
+ignore noexec ${HOME}
+ignore noexec /tmp
+
+blacklist /tmp/.X11-unix
+blacklist ${RUNUSER}
+
+noblacklist ${HOME}/.cargo/credentials
+noblacklist ${HOME}/.cargo/credentials.toml
+
+# Allows files commonly used by IDEs
+include allow-common-devel.inc
+
+# Allow lua (blacklisted by disable-interpreters.inc)
+#include allow-lua.inc
+
+# Allow perl (blacklisted by disable-interpreters.inc)
+#include allow-perl.inc
+
+# Allow python (blacklisted by disable-interpreters.inc)
+#include allow-python2.inc
+#include allow-python3.inc
+
+# Allow ssh (blacklisted by disable-common.inc)
+#include allow-ssh.inc
+
+include disable-common.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-xdg.inc
+
+#mkdir ${HOME}/.cargo
+#whitelist ${HOME}/YOUR_CARGO_PROJECTS
+#whitelist ${HOME}/.cargo
+#whitelist ${HOME}/.rustup
+#include whitelist-common.inc
+include whitelist-runuser-common.inc
+include whitelist-usr-share-common.inc
+include whitelist-var-common.inc
+
```

Comment:
FWIW I tested apparmor support. All it takes is a comment here and a few additional lines in /etc/apparmor.d/local/firejail-default. Maybe I'm overdoing it, what do you think?

Suggestion
```
# NOTE: to enable apparmor you will need to uncomment and edit
# the 'cargo + rustup' rules in /etc/apparmor.d/local/firejail-default.
# Then you can add 'apparmor' to your cargo.local.
#apparmor
```

For [firejail-local](https://github.com/netblue30/firejail/blob/master/etc/apparmor/firejail-local):
```
## Uncomment to opt-in to apparmor for 'cargo + rustup'.
#+ If you use whitelisting change ${HOME}/YOUR_CARGO_PROJECTS below to your Rust projects directory.
#+ Otherwise you can keep that line commented.
#owner @{HOME}/.cargo/bin/*/** ix,
#owner @{HOME}/.cargo/env ix,
#owner @{HOME}/.rustup/toolchains/*/** ix,
#owner @{HOME}/YOUR_CARGO_PROJECTS/** ix,
```

Reply:
`cargo install` needs `/tmp` mounted `exec` IIRC. Just saying, I don't even know if `apparmor` makes it `noexec`.

Reply:
More testing confirms what you're saying. AA support is out.